srml/balances/src/lib.rs

```diff
@@ -662,16 +643,21 @@ where
 		who: &T::AccountId,
 		value: Self::Balance,
 	) -> Self::PositiveImbalance {
-		Self::ensure_free_balance_is(who, Self::free_balance(who) + value);
-		PositiveImbalance(value)
+		let (imbalance, _) = Self::ensure_free_balance_is(who, Self::free_balance(who) + value);
```

Comment:
I still don't think this protects against `value < ed` for an account that doesn't exist when creating an account.

Example:
ed = 10
value = 5
```
let original = Self::free_balance(who); // original = 0
let imbalance = if original <= balance { // true: 0 < 5
	SignedImbalance::Positive(PositiveImbalance(balance - original)) // imbalance = SignedImbalance::Positive(PositiveImbalance(5))
}

let outcome = if balance < <Module<T, I>>::existential_deposit() { // true: 5 < 10
	Self::set_free_balance(who, balance); // <-- does not handle imbalance
	UpdateBalanceOutcome::AccountKilled // outcome
}

// return
(imbalance, outcome) // (SignedImbalance::Positive(PositiveImbalance(5)), UpdateBalanceOutcome::AccountKilled)
```
This will tell the calling function to add 5 units to issuance but will not create the account with 5 units in it.

Reply:
OK, seems to be handled in `on_free_too_low` (called by `set_free_balance`)
```
if !dust.is_zero() {
	T::DustRemoval::on_unbalanced(NegativeImbalance(dust));
}
```

Reply:
This still works. The imbalance will indeed be positive, which will cancel out dust cleanup called in `set_free_balance`.

However, there is an edge case, which is where there are no other balances on the system. This wouldn't happen in practice, but I'll add a workaround for it anyway. Also a couple of tests.